src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -1270,15 +1270,20 @@ def visit_If(self, node: ast.If) -> list:
         return result
 
     def visit_While(self, node: ast.While) -> gt_ir.While:
+        loc = gt_ir.Location.from_ast_node(node)
         if node.orelse:
             raise GTScriptSyntaxError("orelse is not supported on while loops")
         stmts = []
         for stmt in node.body:
             stmts.extend(self.visit(stmt))
+            if isinstance(stmts[-1], gt_ir.While):
+                raise GTScriptSyntaxError(
+                    "Nested while loops detected. This is unsupported.", loc=loc
+                )
```

Comment:
Please add a test, it's currently not working.

The fix is to either change line 1278 and check if `self.visit(stmt)` is a list or not or return a list from this method. I am not sure which is the right solution as I am not familiar with the frontend.

Reply:
After you mentioned that, I took another look. There's nothing blocking nested while loop, and I actually checked that it works correctly in gtcpp and the new gtc:numpy backend, so I removed the check here and cleaned it up.

Reply:
Looks good. The test that I had is now working.